ext/gd/gd.c

```diff
@@ -1878,7 +1925,34 @@ PHP_FUNCTION(imagepng)
 /* {{{ Output WEBP image to browser or file */
 PHP_FUNCTION(imagewebp)
 {
-	_php_image_output_ctx(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_GDIMG_TYPE_WEBP, "WEBP");
+	zval *imgind;
+	zend_long quality = -1;
+	gdImagePtr im;
+	gdIOCtx *ctx;
+	zval *to_zval = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!l", &imgind, gd_image_ce, &to_zval, &quality) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	im = php_gd_libgdimageptr_from_zval_p(imgind);
+
+	ctx = create_output_context(to_zval);
+	if (!ctx) {
+		RETURN_FALSE;
+	}
+
+	if (quality < -1) {
+		zend_argument_value_error(3, "must be greater than or equal to -1");
+		ctx->gd_free(ctx);
+		RETURN_THROWS();
+	}
```

Comment:
This check should actually be before our call to `php_gd_libgdimageptr_from_zval_p()`

Reply:
I put them in this order to have the checks in the same order of the parameters, so if you have a two bad parameters you first get the error from the leftmost one. It seems logical to me...

Reply:
Ideally yes, but we already don't do this in other extensions, because if one needs to allocate something it is pointless to do that work just to throw it away because of a programming error.

Reply:
We are still talking about an error path, even if we allocate to then deallocate it's not going to impact in any way, and I value the logical order of things.

Either way, this is still a refactor and I restored the exact behavior that was already in place, so any functional change should be done in a follow up commit.